src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -110,73 +110,71 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
+
+            // Look for exact matches first:
+            var exactReferences = await FindResultsAsync(projectToAssembly, referenceToCompilation, project, finder, exact: true, cancellationToken: cancellationToken).ConfigureAwait(false);
+            if (exactReferences?.Count > 0)
+            {
+                return exactReferences;
+            }
+
+            // No exact matches found.  Fall back to fuzzy searching.
+            var fuzzyReferences = await FindResultsAsync(projectToAssembly, referenceToCompilation, project, finder, exact: false, cancellationToken: cancellationToken).ConfigureAwait(false);
+            if (fuzzyReferences?.Count > 0)
+            {
+                return fuzzyReferences;
+            }
+
+            return null;
+        }
+
+        private async Task<List<Reference>> FindResultsAsync(
             ConcurrentDictionary<Project, AsyncLazy<IAssemblySymbol>> projectToAssembly,
             ConcurrentDictionary<PortableExecutableReference, Compilation> referenceToCompilation,
-            Project project, List<SymbolReference> allSymbolReferences, SymbolReferenceFinder finder, bool exact, CancellationToken cancellationToken)
+            Project project, SymbolReferenceFinder finder, bool exact, CancellationToken cancellationToken)
         {
+            var allReferences = new List<Reference>();
+
             // First search the current project to see if any symbols (source or metadata) match the 
             // search string.
-            await FindResultsInAllProjectSymbolsAsync(project, allSymbolReferences, finder, exact).ConfigureAwait(false);
+            await FindResultsInAllProjectSymbolsAsync(project, allReferences, finder, exact).ConfigureAwait(false);
 
             // Now search unreferenced projects, and see if they have any source symbols that match
             // the search string.
-            await FindResultsInUnreferencedProjectSourceSymbolsAsync(projectToAssembly, project, allSymbolReferences, finder, exact, cancellationToken).ConfigureAwait(false);
+            await FindResultsInUnreferencedProjectSourceSymbolsAsync(projectToAssembly, project, allReferences, finder, exact, cancellationToken).ConfigureAwait(false);
+
+            // We only support searching NuGet in an exact manner currently. 
+            if (exact)
+            {
+                await finder.FindNugetReferencesAsync(allReferences).ConfigureAwait(false);
+            }
 
             // Finally, check and see if we have any metadata symbols that match the search string.
-            await FindResultsInUnreferencedMetadataSymbolsAsync(referenceToCompilation, project, allSymbolReferences, finder, exact, cancellationToken).ConfigureAwait(false);
+            await FindResultsInUnreferencedMetadataSymbolsAsync(referenceToCompilation, project, allReferences, finder, exact, cancellationToken).ConfigureAwait(false);
```

Comment:
That won't be a problem when your other fix goes in to prevent exactly that, right? At which point, if some other project in your solution references an assembly (which is NOT contained in a package) containing Foo, but also there's a NuGet package out there *that you've never referenced before* containing Foo, would you still prefer the NuGet package? Or is there a more compelling example in the other direction?